utils/measure/measure/powermeter/hass.py

```diff
@@ -52,5 +74,26 @@ def get_power_sensors(self) -> list[str]:
         power_sensors = [entity.entity_id for entity in sensors if entity.state.attributes.get("unit_of_measurement") == "W"]
         return sorted(power_sensors)
 
+    def get_voltage_sensors(self) -> list[str]:
+        entities = self.client.get_entities()
+        sensors = entities["sensor"].entities.values()
+        voltage_sensors = [entity.entity_id for entity in sensors if entity.state.attributes.get("unit_of_measurement") == "V"]
+        return sorted(voltage_sensors)
+
+    def match_power_and_voltage_sensors(self) -> dict[str, str]:
+        power_sensors = self.get_power_sensors()
+        voltage_sensors = self.get_voltage_sensors()
+
+        # Create mappings based on base names
+        power_map = {sensor.rsplit("_power", 1)[0]: sensor for sensor in power_sensors}
```

Comment:
This logic is quite prone to errors. we could also just ask the user to select the voltage entity, by adding a question. You can do to in the `get_questions` method.
You can also add a lamba here to check if `dummy_load` has been selected, when it's not we can omit this question.
There are some examples in other parts of the code.

Reply:
Yes I know. But the only other solution I came up with is asking the REST-API of HomeAssistant for devices, as devices are not exposed via the ha library at the moment.

But I mean the worst case scenario would only occur if the user renamed the voltage or power meter entity-id. This seemed pretty unlikely to me.

I thought the better call is to use this and wait for the ha library to catch up and offer device listings.

But if you think that's not good enough we could do this current matching, and if no match is found the user is asked for the entity id of the voltage meter from the list.

Reply:
> You can also add a lamba here to check if `dummy_load` has been selected, when it's not we can omit this question.

Well, the mapping is only searched if get_power() is called with include_voltage, otherwise it's omitted anyway.

https://github.com/bramstroker/homeassistant-powercalc/blob/2edbbc2e20beea3c390ea5d1ca3a097ee0e67e87/utils/measure/measure/powermeter/hass.py#L40

Reply:
They won't implement it in the REST API ever to get devices, I also missed that a few times when I needed it.
If I remember correctly I even made a PR in HA core to get it added.
But the REST API is kinda deprecated, they won't add any new features there.

Yes we could first try to find by convention, and when non can be found we can promt the question at runtime.
Then we have best of both worlds.

Reply:
https://github.com/home-assistant/core/pull/57437

Reply:
Yes that would certainly be nice, to keep the same pattern everywhere a question is prompted